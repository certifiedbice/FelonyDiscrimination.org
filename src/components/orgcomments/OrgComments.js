import React,{Component} from 'react'
import OrgApiService from '../../services/org-api-service';
import './OrgComments.css';

export default class OrgComments extends Component{
    state={setError:null};
    componentDidMount(){
        //set comments in state for render
    }
    handleSubmit=e=>{
        e.preventDefault()
        const {comment}=e.target;
        // Have to add in userId when user login system is implemented.
        OrgApiService.postOrgComments(/*this.props.userId,*/this.props.orgId,comment)
            .catch(this.state.setError);
    }
    render(){
        return(
            <div id='organization-comments-container'>
                <div className="comment-list-container">

                </div>
                <div className="organization-comment-form-container">
                    <form id="comment-form" name="comment-form" aria-label="Comment form" onSubmit={this.handleSubmit}>
                        <legend>Submit a Comment</legend>
                        <div className="comment-form-element-container">
                            <label>Message:</label>
                            <textarea className="comment-form-element" name="comment" id="comment-form-message" rows="6" cols="30" required aria-labelledby="review-form-message"/>
                        </div>
                    </form>
                </div>
            </div>
        );
	}
}