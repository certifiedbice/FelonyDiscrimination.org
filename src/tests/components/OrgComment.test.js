import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import OrgComments from '../../components/orgcomments/OrgComments';

let orgId='5';

it('renders without crashing',()=>{
    const div=document.createElement('div');
    ReactDOM.render(<OrgComments orgId={orgId}/>,div);
});

it('renders correctly',()=>{
    const tree=renderer
        .create(<OrgComments orgId={orgId}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
});