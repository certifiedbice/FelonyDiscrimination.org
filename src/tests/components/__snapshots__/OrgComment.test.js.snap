// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="organization-comments-container"
>
  <div
    className="comment-list-container"
  />
  <div
    className="organization-comment-form-container"
  >
    <form
      aria-label="Comment form"
      id="comment-form"
      name="comment-form"
      onSubmit={[Function]}
    >
      <legend>
        Submit a Comment
      </legend>
      <div
        className="comment-form-element-container"
      >
        <label>
          Message:
        </label>
        <textarea
          aria-labelledby="review-form-message"
          className="comment-form-element"
          cols="30"
          id="comment-form-message"
          name="comment"
          required={true}
          rows="6"
        />
      </div>
    </form>
  </div>
</div>
`;
